### PR TITLE
Processing data models

### PR DESCRIPTION
**Pull request by LeslieAine**
In this PR, I added associations between tables using Rails models.

User

- has_many comments
- has_many posts
- has_many likes

Post

- belongs_to user
- has_many comments
- has_many likes

Like

- belongs_to user
- belongs_to post

Comments

- belongs_to user
- belongs_to post

I also created some methods for querying data for each of these classes.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_post_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (gemmen29) [inline app/models/post.rb:8]: - [ ] [OPTIONAL] Please use the callback `after_save` to call the `update_post_counter`, so the method is called automatically every time a new post has been saved to the database, then make the `update_post_counter` method private.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (gemmen29) [inline app/models/like.rb:7]: - [ ] [OPTIONAL] Please use the callback `after_save` to call the `update_likes_counter`, so the method is called automatically every time a new post has been saved to the database, then make the `update_likes_counter` method private.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (gemmen29) [inline app/models/comment.rb:7]: - [ ] [OPTIONAL] Please use the callback `after_save` to call the `update_comments_counter`, so the method is called automatically every time a new post has been saved to the database, then make the `update_comments_counter` method private.

**Review by gemmen29 — APPROVED** [review]
Hi @LeslieAine, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions:
- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 3 above)
Comment (LeslieAine) [reply app/models/comment.rb:7]: @gemmen29 thank you for the feedback. It is very helpful

**Pull request merged**